runtime/fresh/routes/invoke.ts

```diff
@@ -1,11 +1,41 @@
 import type { HandlerContext } from "$fresh/server.ts";
+import { formDataToProps } from "deco/clients/formdata.ts";
 import type { DecoSiteState, DecoState } from "../../../types.ts";
 import { bodyFromUrl } from "../../../utils/http.ts";
 import { invokeToHttpResponse } from "../../../utils/invoke.ts";
 import type { InvokeFunction } from "../../../utils/invoke.types.ts";
 
 import { payloadToResolvable, wrapInvokeErr } from "./batchInvoke.ts";
 
+const PROPS_FROM_REQUEST_STRATEGIES = {
+  "json": async (req: Request) => await req.json() as Record<string, any>,
+  "form-data": async (req: Request) => {
+    const formData = await req.formData();
+    return formDataToProps(formData);
+  },
+};
+
+async function parsePropsFromRequest(
+  req: Request,
+): Promise<Record<string, any>> {
+  const contentType = req.headers.get("content-type");
+  const contentLength = req.headers.get("content-length");
+
+  if (contentLength === "0") {
+    return {};
+  }
+
+  if (contentType === "application/json") {
```

Comment:
API-wise (not considering the client that is making the call) this is a breaking change, right? you can use `"Content-Type":"application/json; charset=utf-8",` (which is very common) and then it will be parsed as `{}` , should it defaults to json instead?

I think the best way is to have one `content-type` parser which returns the actual content-type of the given body and a map from `content-type` to `bodyParser`. Otherwise fallsback to json() or {} (if content-length is not present or zero)